### PR TITLE
Yandex.Browser version 23.1.3.952

### PR DESCRIPTION
**Pull request by Andrej730**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/97827)

### CODE REVIEW

**Comment by Andrej730** [comment]
@microsoft-github-policy-service agree

File: manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.4.0.schema.json
+
+PackageIdentifier: Yandex.Browser
+PackageVersion: 23.1.3.952
```
Comment (mdanish-kh) [inline manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml:5]:
```suggestion
PackageVersion: 23.1.3.949
```
Must match Control Panel / Apps&Features

File: manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json
+
+PackageIdentifier: Yandex.Browser
+PackageVersion: 23.1.3.952
```
Comment (mdanish-kh) [inline manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.locale.en-US.yaml:5]:
```suggestion
PackageVersion: 23.1.3.949
```
Must match Control Panel / Apps&Features

File: manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.4.0.schema.json
+
+PackageIdentifier: Yandex.Browser
+PackageVersion: 23.1.3.952
```
Comment (mdanish-kh) [inline manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.yaml:5]:
```suggestion
PackageVersion: 23.1.3.949
```
Must match Control Panel / Apps&Features

File: manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.locale.en-US.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json
+
+PackageIdentifier: Yandex.Browser
+PackageVersion: 23.1.3.952
+PackageLocale: en-US
+Publisher: YANDEX LLC
```
Comment (mdanish-kh) [inline manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.locale.en-US.yaml:7]:
```suggestion
Publisher: YANDEX
```
Must match Control Panel / Apps&Features

File: manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.locale.en-US.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json
+
+PackageIdentifier: Yandex.Browser
+PackageVersion: 23.1.3.952
+PackageLocale: en-US
+Publisher: YANDEX LLC
+PackageName: Yandex Browser
```
Comment (mdanish-kh) [inline manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.locale.en-US.yaml:8]:
```suggestion
PackageName: Yandex
```

File: manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.4.0.schema.json
+
+PackageIdentifier: Yandex.Browser
```
Comment (mdanish-kh) [inline manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml:4]:
```suggestion
PackageIdentifier: Yandex.Yandex
```
WinGet generally follows the naming scheme `Publisher.PackageName` for the PackageID.

File: manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.locale.en-US.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json
+
+PackageIdentifier: Yandex.Browser
```
Comment (mdanish-kh) [inline manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.locale.en-US.yaml:4]:
```suggestion
PackageIdentifier: Yandex.Yandex
```
WinGet generally follows the naming scheme `Publisher.PackageName` for the PackageID.

File: manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.yaml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# Created using wingetcreate 1.2.5.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.4.0.schema.json
+
+PackageIdentifier: Yandex.Browser
```
Comment (mdanish-kh) [inline manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.yaml:4]:
```suggestion
PackageIdentifier: Yandex.Yandex
```
WinGet generally follows the naming scheme `Publisher.PackageName` for the PackageID.

File: manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+PackageVersion: 23.1.3.952
+InstallerSuccessCodes:
+- 1
+- 3
+Installers:
+- Architecture: x64
+  InstallerType: exe
+  InstallerUrl: https://cachev2-ams01.cdn.yandex.net/download.cdn.yandex.net/browser/int/23_1_3_949_40687/en/Yandex.exe
+  InstallerSha256: 0159a0da24910b6b9cbdf888f82862e76bdc334d467f0156c279a82fe55eb449
```
Comment (mdanish-kh) [inline manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml:13]:
```suggestion
  InstallerSha256: 0159a0da24910b6b9cbdf888f82862e76bdc334d467f0156c279a82fe55eb449
  ProductCode: YandexBrowser
```
For better upgrade matching

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
After committing these changes, you would also have to modify the directory structure to match the new PackageIdentifier and PackageVersion. `manifests/y/Yandex/Browser/23.1.3.952` would become `manifests/y/Yandex/Yandex/23.1.3.949`. Subsequently, the .yaml file names would also need updating to match the new PackageId. (Yandex.Browser.installer.yaml -> Yandex.Yandex.yaml and so on..)

A quick and easy way to do this would be here in GitHub Web by pressing the <kbd>.</kbd> (dot) key on your keyboard that'll open up the PR in VS Code Online. Navigate to Explorer menu, find the required files and make the appropriate changes.

After that, stage and commit your changes from Source Control menu on the left. (The changes are automatically pushed)

File: manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.locale.en-US.yaml (hunk 7 above)
Comment (Andrej730) [reply manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.locale.en-US.yaml:4]: @mdanish-kh 
I think `Yandex.Browser` would be much more recognizable for users than just `Yandex.Yandex`
`Yandex.Yandex` can be easily confused with just yandex search bar widget or something else.

Comment (mdanish-kh) [reply manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.locale.en-US.yaml:4]: Sure, that isn't a strict rule anyways. But the metadata should be matching `appwiz.cpl` for upgrade matching's sake

File: manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml (hunk 9 above)
Comment (Andrej730) [reply manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml]: If `productcode` is defined then upgrade will able to find the app to uninstall before installing new version?

Comment (mdanish-kh) [reply manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml]: No, ProductCode just defines the registry key where the installer writes its uninstall entries and related metadata for the package. It is one of the heuristics WinGet uses to find a match for a package.

The behavior you described can be defined by setting `UpgradeBehavior: uninstallPrevious` in the manifest for packages that install side-by-side. Most packages usually upgrade gracefully by just calling the installer.

Comment (Andrej730) [reply manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml]: How do you figure `ProductCode` for some specific package?

Comment (mdanish-kh) [reply manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml]: For MSI Installers, you can get it by running this command in Windows PowerShell

`get-wmiobject Win32_Product | Sort-Object Name | Format-Table IdentifyingNumber, Name -AutoSize`

For other installers, you would have to manually search the registry or use some PowerShell scripting to get the uninstall keys. I would recommend using [SandboxTest.ps1](https://github.com/microsoft/winget-pkgs/blob/master/Tools/SandboxTest.ps1) script before submitting a manifest that will display all Registry entries for you.

**Review by mdanish-kh — APPROVED** [review]

**Comment by mdanish-kh** [comment]
@Andrej730 Please mark all the above conversations as resolved. They are blocking the auto-merge.

File: manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml (hunk 9 above)
Comment (Andrej730) [reply manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml]: Is there some way to get ProductCode from WinGetSvc-Validation logs when the package gets validated on Azure servers?
Don't have Windows Sandbox on my system :\

Comment (mdanish-kh) [reply manifests/y/Yandex/Browser/23.1.3.952/Yandex.Browser.installer.yaml]: @Andrej730 In that case, you can resort to using a PowerShell script to retrieve information about the installed softwares on your machine. Some google searches led to me to this article https://powershell.one/code/5.html, the property that you're looking for is `PSChildName`

**Pull request merged**